domain/learningandworkprogress/src/test/kotlin/uk/gov/justice/digital/hmpps/domain/learningandworkprogress/review/service/ReviewScheduleDateCalculationServiceWithEarliestScheduleDateTest.kt

```diff
@@ -0,0 +1,323 @@
+package uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.service
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.ReviewScheduleCalculationRule
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.ReviewScheduleWindow
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.SentenceType
+import uk.gov.justice.digital.hmpps.domain.randomValidPrisonNumber
+import java.time.LocalDate
+import java.util.stream.Stream
+
+/**
```

Comment:
New temporary test class. As per the doc it tests the date calc service when the earliest date from which all calculated schedules has been specified (ie. 1st April)